pgx-pg-sys/src/submodules/thread_check.rs

```diff
@@ -0,0 +1,82 @@
+//! Enforces thread-safety in `pgx`.
+//!
+//! It's UB to call into Postgres functions from multiple threads. We handle
+//! this by remembering the first thread to call into postgres functions, and
+//! panicking if we're called from a different thread.
+//!
+//! This is called from the current crate from inside the setjmp shim, as that
+//! code is *definitely* unsound to call in the presense of multiple threads.
+//!
+//! This is somewhat heavyhanded, and should be called from fewer places in the
+//! future...
+
+use std::num::NonZeroUsize;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+static ACTIVE_THREAD: AtomicUsize = AtomicUsize::new(0);
+#[track_caller]
+pub(crate) fn check_active_thread() {
+    let current_thread = nonzero_thread_id();
+    // Relaxed is sufficient as we're only interested in the effects on a single
+    // atomic variable, and don't need synchronization beyond that.
+    return match ACTIVE_THREAD.load(Ordering::Relaxed) {
```

Comment:
Unless there's a specific reason that this final expression has an explicit return?
```suggestion
    match ACTIVE_THREAD.load(Ordering::Relaxed) {
```

Reply:
Oh, it used to have the helper functions defined locally in that function, at the bottom after the return. Nice catch.

Reply:
I'll add this to the batch commit but it'll deffo fail afterwards since there's a semicolon on the end.